modules/ballerina-core/src/main/java/org/ballerinalang/model/util/JsonParser.java

```diff
@@ -0,0 +1,1159 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.model.util;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.ballerinalang.model.util.JsonNode.Type;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+
+/**
+ * This class represents a JSON parser.
+ * 
+ * @since 0.95.5
+ */
+public class JsonParser {
```

Comment:
Why do we need a JSON Parser? I thought ANTLR already parse JSON.

Reply:
Sanjiva' initial idea was to parse JSON strings using the ANTLR parser. 

e.g. you can write grammar just for JSON and maintain a small JSON Parser using ANTLR. that will eliminate the need for a JSON parser. Not sure about the performance.

Reply:
At the moment, it's a simple parser that's there, character escaping and so on hasn't been implemented there, it is done by the other JSON parser. Also, we can't use the same exact code, we will have to have separate grammar file and write code for that separately. Also, performance would be slower for sure, where this would be a critical aspect when parsing large or many JSON content in service requests and so on.